src/integrations/unity_cloud_integration.rs

```diff
@@ -0,0 +1,195 @@
+use errors::UnityRetrievalError;
+use failure::Error;
+use integrations::unity_cloud_response::*;
+use network::{get_basic_credentials, get_url_response};
+use remote_status::RemoteStatus;
+use reqwest::header::{Authorization, ContentType, Headers};
+use std::time::Duration;
+use std::time::Instant;
+use RemoteIntegration;
+
+const UNITY_SLEEP_DURATION: u64 = 1000 * 60;
+
+pub struct UnityCloudIntegration {
+    r: u16,
+    g: u16,
+    b: u16,
+    api_token: String,
+    base_url: String,
+    last_tick: Instant,
+    last_status: RemoteStatus,
+}
+
+impl UnityCloudIntegration {
+    pub fn new(r: u16, g: u16, b: u16, api_token: &str, base_url: &str) -> UnityCloudIntegration {
+        UnityCloudIntegration {
+            r: r,
+            g: g,
+            b: b,
+            api_token: api_token.to_string(),
+            base_url: base_url.to_string(),
+            last_tick: Instant::now() - Duration::from_secs(60),
+            last_status: RemoteStatus::Unknown,
+        }
+    }
+
+    fn get_status_internal(&self) -> Vec<Result<(UnityBuildStatus, Headers), UnityRetrievalError>> {
```

Comment:
`get_status_internal` was moved here from `main.rs`, and used to be called `run_one_unity`. It underwent some minor changes in the process. Its helper methods came along for the ride.